src/main/java/com/hms_networks/americas/sc/thingworx/TWConnectorMain.java

```diff
@@ -88,14 +89,13 @@ private static void runGrabData() {
 
         // Check if queue is behind
         try {
-          long queueBehindMillis =
-              HistoricalDataQueueManager.getCurrentTimeWithOffset()
-                  - HistoricalDataQueueManager.getCurrentTimeTrackerValue();
+          long queueBehindMillis = HistoricalDataQueueManager.getQueueTimeBehindMillis();
           if (queueBehindMillis >= TWConnectorConsts.QUEUE_DATA_POLL_BEHIND_MILLIS_WARN) {
+            String timeBehindString =
+                SCTimeUtils.getDayHourMinSecsForMillis(
+                    (int) queueBehindMillis, "days", "hours", "minutes", "seconds");
```

Comment:
i feel like this method should be overloaded to have a default that doesnt take these strings. seems unnecessary as all our connectors dont make any effort for any other type of localization.  

SCTimeUtils.getDayHourMinSecsForMillis(behindMillis)

Reply:
if not these strings should be finals somewhere rather than just defined in the call.

Reply:
Created constants for these. I was conflicted during the creation of the SCTimeUtils.getDayHourMinSecsForMillis() method in the library because of the localization point you made, but I also figured that it is probably best to allow the application to handle strings and localization on its own (if desired), similar to how we used custom exceptions in many places instead of static logs.

Reply:
we could just add SCTimeUtils.getDayHourMinSecsForMillisENUS(millis)